### PR TITLE
Update Node to v16

### PR DESCRIPTION
**Pull request by chiaramooney**
### Type of Change
- Version Upgrade

### Why
Upgrade Node to v16. Upstream is making the same update for v72.

Resolves https://github.com/microsoft/react-native-windows/issues/11451
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/11500)

### CODE REVIEW

**Comment by asklar** [comment]
doesn't the setup script need to be updated too?

File: vnext/Scripts/rnw-dependencies.ps1 [hunk 1]
```diff
@@ -412,7 +412,7 @@ $requirements = @(
     },
     @{
         Id=[CheckId]::Node;
-        Name = 'Node.js (LTS, >= 14.0)';
+        Name = 'Node.js (LTS, >= 16.0)';
```
Comment (jonthysell) [inline vnext/Scripts/rnw-dependencies.ps1:415]: You also need to update the CheckNode function

File: .ado/jobs/node-tests.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ parameters:
 
   - name: versions
     type: object
-    default: [14, 16]
+    default: [16]
```
Comment (jonthysell) [inline .ado/jobs/node-tests.yml:14]: We might want to add node 18 in here since it's been LTS for a long time now

**Review by jonthysell — APPROVED** [review]
Once you at least update the dep script it should be good to go.

File: .ado/jobs/node-tests.yml (hunk 2 above)
Comment (vmoroz) [reply .ado/jobs/node-tests.yml:14]: Node.js is about to release version 20.

**Review by vmoroz — APPROVED** [review]

File: .ado/jobs/node-tests.yml (hunk 2 above)
Comment (jonthysell) [reply .ado/jobs/node-tests.yml:14]: We should definitely add 20 when it's out and see if there's any problems as new devs will probably install 20 by default

**Review by jonthysell — APPROVED** [review]

File: .ado/jobs/node-tests.yml (hunk 2 above)
Comment (chiaramooney) [reply .ado/jobs/node-tests.yml:14]: Addition of 18 threw errors in the testing. Looks like there's an API that is no longer available on 18 that we use. I'll file an issue to track addressing that.

Comment (jonthysell) [reply .ado/jobs/node-tests.yml:14]: You know, that sounds familliar - something that changed at like 18.5 or something.

**Review by jonthysell — APPROVED** [review]

**Comment by jacdebug** [comment]
Bump babel config as well? 
https://github.com/microsoft/react-native-windows/blob/b4e4a90e184cd2f20dc03d87b79f87185670f4e5/packages/%40rnw-scripts/babel-node-config/babel.config.js#L14

**Pull request merged**